@xen-orchestra/lite/src/components/Table.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import styled from 'styled-components'
+import { withState } from 'reaclette'
+
+import IntlMessage from './IntlMessage'
+
+export type Column<Type> = {
+  header: React.ReactNode
+  id?: string | number
+  render: { (item: Type): string | JSX.Element }
+}
+
+export type Item<T> = T & {
+  id?: string
+}
+
+interface ParentState {}
+
+interface State {}
+
+interface Props {
+  collection: Item<unknown>[] | undefined
+  columns: Column<any>[]
+}
+
+interface ParentEffects {}
+
+interface Effects {}
+
+interface Computed {}
+
+const StyledTable = styled.table`
+  border: 1px solid #333;
+  td {
+    border: 1px solid #333;
+  }
+  thead {
+    background-color: #333;
+    color: #fff;
+  }
+`
+const Table = withState<State, Props, Effects, Computed, ParentState, ParentEffects>({}, ({ collection, columns }) =>
+  collection !== undefined && collection.length !== 0 ? (
+    <StyledTable>
+      <thead>
+        <tr>
+          {columns.map((col, index) => (
+            <td key={col.id ?? index}>{col.header}</td>
+          ))}
+        </tr>
+      </thead>
+      <tbody>
+        {collection.map((item, index) => (
+          <tr key={item.id ?? index}>
+            {columns.map((col, index) => (
+              <td key={col.id ?? index}>{col.render(item)}</td>
+            ))}
+          </tr>
+        ))}
+      </tbody>
+    </StyledTable>
+  ) : (
+    <IntlMessage id='noData' />
```

Comment:
We should probably differentiate between `undefined` (no data, ie still loading, error, etc.) and `[]` (the data is there and it's an empty collection).